wandb/sdk/wandb_artifacts.py

```diff
@@ -1175,61 +1153,122 @@ class WBArtifactHandler(StorageHandler):
     def __init__(self, scheme=None):
         self._scheme = scheme or "wandb-artifact"
         self._cache = get_artifacts_cache()
+        self._client = None
 
     @property
     def scheme(self):
+        """overrides parent scheme
+        :return: The scheme to which this handler applies.
+        :rtype: str
+        """
         return self._scheme
 
     @staticmethod
     def parse_path(path):
+        """helper method to parse a URL into it's component parts."""
+
         url = urlparse(path)
         return url.netloc, (url.path if url.path[0] != "/" else url.path[1:])
 
+    @property
+    def client(self):
+        if self._client is None:
+            self._client = PublicApi()
+        return self._client
+
     def load_path(self, artifact, manifest_entry, local=False):
+        """
+        Loads the file within the specified artifact given its
+        corresponding entry. In this case, the referenced artifact is downloaded
+        and a new symlink is created and returned to the caller.
+
+        :param manifest_entry: The index entry to load
+        :type manifest_entry: ArtifactManifestEntry
+        :return: A path to the file represented by `index_entry`
+        :rtype: os.PathLike
+        """
+
+        # Standard shortcircut if the asset is already downloaded
         path, hit = self._cache.check_etag_obj_path(
             manifest_entry.digest, manifest_entry.size
         )
         if hit:
             return path
 
+        # Parse the reference path and download the artifact if needed
         artifact_id, artifact_file_path = WBArtifactHandler.parse_path(
             manifest_entry.ref
         )
-        artifact = PublicApi().artifact_from_id(util.hex_to_b64_id(artifact_id))
-        artifact_path = artifact.download()
 
-        link_target_path = os.path.join(artifact_path, artifact_file_path)
+        dep_artifact = PublicArtifact.from_id(
+            util.hex_to_b64_id(artifact_id), self.client
+        )
+        link_target_path = dep_artifact.get_path(artifact_file_path).download()
         link_creation_path = os.path.join(
```

Comment:
I think we discussed getting rid of this right? And getting rid of symlinking entirely for now?

Reply:
Got rid of it entirely in the next commit.